### PR TITLE
LANG-1659 - Add null-safe ObjectUtils.isArray()

### PR DESCRIPTION
**Pull request by arturobernalg**
The idea it's have a method that determine whether the given object is an array. Change all the reference of `.getClass().isArray()` in favor of `ObjectUtils.isArray(value)`

### CODE REVIEW

File: src/main/java/org/apache/commons/lang3/ObjectUtils.java [hunk 1]
```diff
@@ -951,6 +951,25 @@ their constant using one of the CONST() utility methods, instead:
...
+     * ObjectUtils.isArray(new int[]{1,2,3}) = true
+     * ObjectUtils.isArray(1234)             = false
+     * </pre>
+     *
+     * @param object the object to check, may be {@code null}
+     * @return {@code true} if the object is an {@code array}, {@code false} otherwise
+     * @since 3.13.0
+     */
+    public static boolean isArray(final Object object) {
+        return (object != null && object.getClass().isArray());
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/ObjectUtils.java:971]: No need  for extra parentheses here.

Comment (arturobernalg) [reply src/main/java/org/apache/commons/lang3/ObjectUtils.java]: Done. Changed

File: src/main/java/org/apache/commons/lang3/ObjectUtils.java [hunk 2]
```diff
@@ -951,6 +951,25 @@ their constant using one of the CONST() utility methods, instead:
             public final static int MAGIC_NUMBER = CONST(5);
      */
 
+    /**
+     * <p>Determine whether the given object is an array:
+     * either an Object array or a primitive array.
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/ObjectUtils.java:956]: Close HTML tags.

File: src/test/java/org/apache/commons/lang3/ObjectUtilsTest.java [hunk 3]
```diff
@@ -586,6 +586,23 @@ public void testIsEmpty() {
...
+    public void testArray() {
+        assertFalse(ObjectUtils.isArray(null));
+        assertFalse(ObjectUtils.isArray(""));
+        assertFalse(ObjectUtils.isArray("abg"));
+        assertFalse(ObjectUtils.isArray(123));
+
+        assertTrue(ObjectUtils.isArray(NON_EMPTY_ARRAY));
+        assertTrue(ObjectUtils.isArray(new int[] {}));
+        assertTrue(ObjectUtils.isArray(new int[]{1, 2, 3}));
+
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/ObjectUtilsTest.java:602]: Remove extra blank lines.

File: src/test/java/org/apache/commons/lang3/ObjectUtilsTest.java [hunk 4]
```diff
@@ -586,6 +586,23 @@ public void testIsEmpty() {
...
+     */
+    @Test
+    public void testArray() {
+        assertFalse(ObjectUtils.isArray(null));
+        assertFalse(ObjectUtils.isArray(""));
+        assertFalse(ObjectUtils.isArray("abg"));
+        assertFalse(ObjectUtils.isArray(123));
+
+        assertTrue(ObjectUtils.isArray(NON_EMPTY_ARRAY));
+        assertTrue(ObjectUtils.isArray(new int[] {}));
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/ObjectUtilsTest.java:600]: Might as well test with all of our own arrays like the `EMPTY_*` arrays.

File: src/main/java/org/apache/commons/lang3/ObjectUtils.java [hunk 5]
```diff
@@ -951,6 +951,25 @@ their constant using one of the CONST() utility methods, instead:
             public final static int MAGIC_NUMBER = CONST(5);
      */
 
+    /**
+     * <p>Determine whether the given object is an array:
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/ObjectUtils.java:955]: I've been using "Tests..." for `is*` methods. If you really want to keep "Determine" then it should be "Determines", also the main part of the Javadoc should mention that the method is null-safe, for example: "Test whether the given object is an Object array or a primitive array in a null-safe manner.".

**Review by garydgregory — CHANGES_REQUESTED** [review]
@arturobernalg 
Please see my scattered comments.

**Comment by arturobernalg** [comment]
HI @garydgregory 
I have made all the changes
TY

File: src/main/java/org/apache/commons/lang3/ObjectUtils.java (hunk 2 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/lang3/ObjectUtils.java]: Done.

File: src/test/java/org/apache/commons/lang3/ObjectUtilsTest.java (hunk 3 above)
Comment (arturobernalg) [reply src/test/java/org/apache/commons/lang3/ObjectUtilsTest.java]: Done

File: src/test/java/org/apache/commons/lang3/ObjectUtilsTest.java (hunk 4 above)
Comment (arturobernalg) [reply src/test/java/org/apache/commons/lang3/ObjectUtilsTest.java]: Done

File: src/main/java/org/apache/commons/lang3/ObjectUtils.java (hunk 5 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/lang3/ObjectUtils.java]: Changed.
TY

**Pull request merged**